### PR TITLE
Add includeProcessVariables in CMMN rest api

### PR DESCRIPTION
**Pull request by sina-al**
#### Check List:
* Unit tests: YES / NO / NA
* Documentation: YES / NO / NA

### CODE REVIEW

File: modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceBaseResource.java [hunk 1]
```diff
@@ -218,6 +218,12 @@
             }
         }
 
+        if (request.getIncludeProcessVariables() != null) {
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceBaseResource.java:221]: I think `request` should be `queryRequest` instead.

File: modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceCollectionResource.java [hunk 2]
```diff
@@ -225,6 +226,10 @@
             queryRequest.setIncludeTaskLocalVariables(Boolean.valueOf(allRequestParams.get("includeTaskLocalVariables")));
         }
 
+        if (requestParams.containsKey("includeProcessVariables")) {
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceCollectionResource.java:229]: Shouldn't `requestParams` be `allRequestParams` instead?

Comment (sina-al) [reply modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceCollectionResource.java]: Indeed 2230030567c3621503a19ff5cfe6eb5f30b56876

File: modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceBaseResource.java (hunk 1 above)
Comment (sina-al) [reply modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/history/task/HistoricTaskInstanceBaseResource.java]: Yes it should 2230030

**Comment by tijsrademakers** [comment]
We don't want to expose this in the CMMN REST API, this is available in the BPMN REST API already.

**Comment by sina-al** [comment]
Hey @tijsrademakers 

What would be the ideal way, via the BPMN REST API, to query for runtime tasks where

- `caseInstanceIdWithChildren` is specified
- `includeProcesVariables` is true

I believe `caseInstanceIdWithChildren` filter is available in on `GET /cmmn-api/cmmn-runtime/tasks`  but on in `GET /bpmn-api/runtime/tasks`

Likewise, I believe `includeProcesVariables` is available in on `GET /bpmn-api/runtime/tasks`  but on in `GET /cmmn-api/cmmn-runtime/tasks`

(at the time this PR was opened)


My current workaround is to do N+1 requests to fit my usecase.

Is there a better approach? Or do you would you say there's something off with this use case in general?

**Comment by tijsrademakers** [comment]
Hi @sina-al,

Thanks for your feedback.
Is it correct that you are looking for a way to get all tasks of a specific case instance where the process tasks from a sub process instance have the process variables included?

Best regards

**Comment by sina-al** [comment]
Hey  @tijsrademakers thanks for the response.

> Is it correct that you are looking for a way to get all tasks of a specific case instance where the process tasks from a sub process instance have the process variables included?

Yeah that's a spot on description of what I'm looking for.

I've also explored inheriting variables on the sub processes instances so that I can filter by them, but when I saw `caseInstanceIdWithChildren` on the `GET /cmmn-api/cmmn-runtime/tasks` endpoint it seemed like the more natural route to go with.

Best regards

**Comment by tijsrademakers** [comment]
Thanks for the additional feedback, I've merged the PR.

**Pull request merged**